### ISSUE TITLE
Chaining framework independent preprocessing defences

### ISSUE THREAD

**Issue by bunni90** [Trusted-AI/adversarial-robustness-toolbox#1888]
**Describe the bug**
When combining multiple defences of which at least one is framework independent art raises the NotImplemented error "The current combination of preprocessing types is not supported.".

**To Reproduce**
1. Import and instantiate mutliple preprocessing defences (at least one framework independent)
`from art.defences.preprocessor import SpatialSmoothingPyTorch #framework specific
from art.defences.preprocessor import FeatureSqueezing #framework independent

spatial_defense = SpatialSmoothingPyTorch(clip_values=[0, 1], channels_first=True)
feature_squeezing = FeatureSqueezing(clip_values=[0, 1])
`
2. Instantiate a (framework specific) classifier with the preprocessing defences as preproseccing_defences
`classifier_def = PyTorchClassifier(
    model=model,
    loss=criterion,
    optimizer=optimizer,
    input_shape=(3, 224, 224),
    nb_classes=1000,
    clip_values=[0, 1],
    preprocessing_defences=[spatial_defense, feature_squeezing]
)`
3. Call predict
`prediction = classifier_def.predict(x)`

**Expected behavior**
Execute the defences on x one after another.

**System information (please complete the following information):**
 - Linux
 - Python 3.8.10
 - ART 1.9 - 1.12
 - TensorFlow / PyTorch
